### DISCUSSION TITLE
How we pass the module options to the runtime plugin/composables... - options have nested properties + function

### DISCUSSION THREAD

**Discussion by nhattran998** [nuxt/nuxt#22744] [Questions] [answered]
I have the module options like this, how do we pass all of this to the runtime?

```ts
type Options = {
	id: string | ((key: string) => string);
    ...xyz: primitive_type;
   onSomething: (xyz) => void;
}
```
In `module.ts` file --> I already use runtimeConfig to pass options to that, but expect the function like `onSomthing...` did not pass.

```ts
    nuxt.options.runtimeConfig.public.XYZ = defu(
      nuxt.options.runtimeConfig.public.XYZ,
      options,
    );
```

So anyone can guide or hint to me on how I can do this with the Nuxt module?
Thanks 🙏 🙇‍♂️

**Comment by nhattran998** [comment, marked as answer]
I find a way to do this like this.

1. Convert your function to string with --> String(function)
2. In runtime, your convert them to function with --> eval(stringFunction)

Example bellow:

```ts
    const { onSomething, ...rest } = options;
    nuxt.options.runtimeConfig.public.XYZ = defu(
      nuxt.options.runtimeConfig.public.XYZ,
      rest,
	 { onSomething: String(onSomthing) }
    );
    
    // In runtime plugin/composables...
  const config = useRuntimeConfig();
  const doSomething = eval(config.public.XYZ.onSomething);

```

**Reply by GaborTorma** [reply]
Is it real? No other way, only the evil?
